src/AppInstallerCommonCore/GroupPolicy.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/GroupPolicy.h"
+#include "AppInstallerLogging.h"
+
+using namespace AppInstaller::StringResource;
+
+namespace AppInstaller::Settings
+{
+    namespace
+    {
+        GroupPolicy& InstanceInternal(std::optional<GroupPolicy*> overridePolicy = {})
+        {
+            // TODO: Use final location
+            constexpr std::string_view PoliciesKeyPath = "test\\policy"; // "SOFTWARE\\Policies\\Microsoft\\Windows\\WinGet";
+            static GroupPolicy s_groupPolicy{ Registry::Key::OpenIfExists(HKEY_CURRENT_USER /* HKEY_LOCAL_MACHINE */, PoliciesKeyPath) };
+            static GroupPolicy* s_override = nullptr;
+
+            if (overridePolicy.has_value())
+            {
+                s_override = overridePolicy.value();
+            }
+
+            return (s_override ? *s_override : s_groupPolicy);
+        }
+
+        template<Registry::Value::Type T>
+        std::optional<decltype(std::declval<Registry::Value>().GetValue<T>())> GetRegistryValue(const Registry::Key& key, const std::string_view valueName)
+        {
+            if (!key)
+            {
+                // Key does not exist; there's nothing to return
+                return std::nullopt;
+            }
+
+            auto value = key[valueName];
+            if (!value.has_value())
+            {
+                return std::nullopt;
+            }
+
+            if (value->GetType() != T)
+            {
+                AICLI_LOG(Core, Warning, << "Value for policy '" << valueName << "' does not have expected type");
+                return std::nullopt;
+            }

```

Comment:
This is restrictive; some of the types are somewhat interchangeable (string vs expand string).  Given that we do not control the code that writes these, we should be more flexible.  I realize that might require catching an exception and continuing, but another option would be to create a TryGetValue method that doesn't throw on a failure.

Reply:
Added a Value::TryGetValue() function